src/server/pkg/obj/testing/client_test.go

```diff
@@ -0,0 +1,282 @@
+		t.Parallel()
+		doWriteTest(t, client, tu.UniqueString("test-single-write-"), []string{"foo"})
+	})
+
+	t.Run("TestMultiWrite", func(t *testing.T) {
+		t.Parallel()
+		doWriteTest(t, client, tu.UniqueString("test-multi-write-"), []string{"foo", "bar"})
+	})
+
+	t.Run("TestSubdirectory", func(t *testing.T) {
+		t.Parallel()
+		object := path.Join(tu.UniqueString("test-subdirectory-"), "object")
+		doWriteTest(t, client, object, []string{"foo", "bar"})
+	})
+
+	// TODO: implement walk test
+
+	t.Run("TestInterruption", func(t *testing.T) {
+		t.Skip("Object client interruption is not currently supported across all clients")
+		// Interruption is currently not implemented on the Amazon, Microsoft, and Minio clients
+		//  Amazon client - use *WithContext methods
+		//  Microsoft client - move to github.com/Azure/azure-storage-blob-go which supports contexts
+		//  Minio client - upgrade to v7 which supports contexts in all APIs
+		t.Parallel()
+
+		// Make a canceled context
+		ctx, cancel := context.WithCancel(context.Background())
+		cancel()
+
+		object := tu.UniqueString("test-interruption-")
+		defer requireExists(t, client, object, false)
+
+		// Some clients return an error immediately and some when the stream is closed
+		w, err := client.Writer(ctx, object)
+		require.NoError(t, err)
+		if w != nil {
+			err = w.Close()
+		}
+		require.YesError(t, err)
+		require.True(t, errors.Is(err, context.Canceled))
+
+		// Some clients return an error immediately and some when the stream is closed
+		r, err := client.Reader(ctx, object, 0, 0)
+		if r != nil {
+			err = r.Close()
+		}
+		require.YesError(t, err)
+		require.True(t, errors.Is(err, context.Canceled))
+
+		err = client.Delete(ctx, object)
+		require.YesError(t, err)
+		require.True(t, errors.Is(err, context.Canceled))
+
+		err = client.Walk(ctx, object, func(name string) error {
+			require.False(t, true)
+			return nil
+		})
+		require.YesError(t, err)
+		require.True(t, errors.Is(err, context.Canceled))
+
+		exists := client.Exists(ctx, object)
+		require.False(t, exists)
+	})
+}
+
+func TestAmazonClient(t *testing.T) {
+	t.Parallel()
+
+	amazonTests := func(t *testing.T, id string, secret string, bucket string, region string, endpoint string) {
+		for _, reverse := range []bool{true, false} {
+			t.Run(fmt.Sprintf("reverse=%v", reverse), func(t *testing.T) {
+				t.Parallel()
+				creds := &obj.AmazonCreds{ID: id, Secret: secret}
+				client, err := obj.NewAmazonClient(region, bucket, creds, "", endpoint, reverse)
+				require.NoError(t, err)
+				runTests(t, client)
+			})
+		}
+	}
+
+	// Test the Amazon client against S3
+	t.Run("AmazonObjectStorage", func(t *testing.T) {
+		t.Parallel()
+		id, secret, bucket, region := LoadAmazonParameters(t)
+		amazonTests(t, id, secret, bucket, region, "")
+	})
+
+	// Test the Amazon client against ECS
+	t.Run("ECSObjectStorage", func(t *testing.T) {
+		t.Parallel()
+		id, secret, bucket, region, endpoint := LoadECSParameters(t)
```

Comment:
Can we deploy ECS inside of CI? Is ECS open-source (or like, generally deployable)?

Or is this just here in case we need to do like one-off testing later?

Reply:
I actually got a developer account with portal.ecstestdrive.com, and this uses the creds for that.